### PR TITLE
Support linking roo-ui to a local app

### LINKED ISSUE #62: Improve development workflow documentation

### PR DESCRIPTION
**Pull request by allanpope**
closes #62

### CODE REVIEW

File: .dev/styled-components/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "styled-components",
+  "version": "0.0.0",
```
Comment (angusfretwell) [inline .dev/styled-components/package.json]: `version` is probably redundant.

File: README.md [hunk 2]
```diff
@@ -90,6 +90,33 @@ Configure [style-loader](https://github.com/webpack-contrib/style-loader) and [c
 
 Also consider configuring [postcss-loader](https://github.com/postcss/postcss-loader) with [autoprefixer](https://github.com/postcss/autoprefixer) if you need vendor prefixes.
 
+## Development workflow
```
Comment (angusfretwell) [inline README.md]: Could we please move this into `CONTRIBUTING.md`, replacing what is currently there about linking

Comment (allanpope) [reply README.md]: Cool, good point

**Comment by angusfretwell** [comment]
@ehyland would be good to get you to try this out with self service and make sure it works for you too 😄

**Comment by ehyland** [comment]
@angusfretwell @allanpope 

Following instructions CONTRIBUTING.md, I've hitting an error when running `yarn workspace @roo-ui/themes link`

```shell
14:11 ~/code/work/hooroo/roo-ui (add-development-workflow) 🥦  yarn workspace styled-components link
yarn workspace v1.6.0
warning Missing version in workspace at "/Users/eamon/code/work/hooroo/roo-ui/.dev/styled-components", ignoring.
error Unknown workspace "styled-components".
info Visit https://yarnpkg.com/en/docs/cli/workspace for documentation about this command.
```

**Comment by angusfretwell** [comment]
@allanpope lol, looks like version is not redundant

**Comment by ehyland** [comment]
@angusfretwell Nah think line 4 or the error with where it's at

> error Unknown workspace "styled-components".

**Comment by angusfretwell** [comment]
@ehyland I think it ignores the package cus it doesn't have a version

**Comment by allanpope** [comment]
@ehyland @angusfretwell Just added the version. The docs said " If you don't plan to publish your package, the name and version fields are optional.". Thought we could get away without the version.

**Comment by ehyland** [comment]
Could we add a script to roo-ui to setup the links

Usage: `./script-name <path-to-consuming-project>`

Script:
```
CONSUMER_PATH="$0"

packages=(
    "styled-components"
    "@roo-ui/components"
    "@roo-ui/fonts"
    "@roo-ui/icons"
    "@roo-ui/logos"
    "@roo-ui/themes"
    "@roo-ui/test-utils"
)

for package in ${packages[@]}; do    
    yarn workspace "$package" link
done

cd "$CONSUMER_PATH"
for package in ${packages[@]}; do    
    yarn link "$package"
    echo "linked $package 🙌"
done
```

**Comment by allanpope** [comment]
@ehyland Yeah awesome idea, i'll add it soon

**Comment by ehyland** [comment]
@allanpope it's working well, reload on save and all.

Would be great it we had a `yarn run build:watch` script in the workspace root, but that's probably out of scope

**Comment by ehyland** [comment]
Implemented `build:watch` [here](https://github.com/hooroo/roo-ui/pull/98)

File: package.json [hunk 3]
```diff
@@ -14,7 +15,9 @@
     "postinstall": "yarn build",
     "build": "yarn lerna run build",
     "storybook": "start-storybook -p 9001 -c .storybook",
-    "deploy-storybook": "storybook-to-ghpages"
+    "deploy-storybook": "storybook-to-ghpages",
+    "link-app": "tasks/link-app.sh",
+    "unlink-app": "tasks/unlink-app.sh"
```
Comment (angusfretwell) [inline package.json]: Just a thought, can we make these `bin/link-app` and `bin/unlink-app` to be a bit more conventional? Also I believe it's preferred to omit the `.sh` file convention, but ensure they have a shebang and `chmod +x`. Same as how our CI scripts are.

File: tasks/link-app.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+packages=($(yarn lerna ls | grep @roo-ui))
+packages+=('styled-components')
+
+yarn lerna exec 'yarn link'
+
+cd $1
+
+for package in ${packages[@]/*v*/}; do
+  yarn link "$package"
+  echo "linked $package 🙌"
```
[the remark this answers is not in the document]
Comment (allanpope) [reply tasks/link-app.sh]: Most of the noise we get currently is from packages not having a license. 
See how we go, we can add this in if needed.

[the remark this answers is not in the document]
Comment (angusfretwell) [reply tasks/link-app.sh]: We got `MIT` in the root should probably propagate that throughout.

**Pull request merged**